### DISCUSSION TITLE
Is it possible to turn off the cache of the translation?

### DISCUSSION THREAD

**Discussion by FiLinX** [FilipePS/Traduzir-paginas-web#544] [Q&A] [answered]
I read that this cache is necessary for the functionality of voicing the voice of the translated text, but I do not use this functionality at all, and at the same time I see that it is constantly growing.
Is it possible to set the cache shutdown for those who do not use voice dubbing? Or at least its auto -cleaning when closing the browser tab?
And thanks for the wonderful tool!

**Comment by FilipePS** [comment]
> I read that this cache is necessary for the functionality of voicing the voice of the translated text,

I don't know where you read this, but this information is not accurate.


The extension has two types of cache:
1. Cache in memory
   - Cleared when closing the browser.
   - used to store text translation
   - used to store the result of text-to-speech
2. Disk cache.
   - Cleared when the extension updates to a new version or you manually clear the cache.
   - Only used for storing text translation (not text-to-speech).


Note that there will only be one entry in the text-to-speech cache when you click the listen to text button (if you don't press it, the cache will have nothing). And when closing the browser this cache is cleared.


>  I see that it is constantly growing.

I believe you entered the configuration page and clicked the button to calculate the cache size, is that it?

The size shown is just for the disk cache only, how big is that cache for you, 500MB?

Why would you want to disable caching?

**Reply by FiLinX** [reply]
> Why would you want to disable caching?

The fact is that I simply do not understand why you need to store these caches at all? And even more so to accumulate them?
I understand that while you are working on the page with the translation involved, yes, the cache may be necessary there. But when you have already left that resource, why store this information about an already completed task?
In addition, I am slightly concerned about data privacy, and wherever I could, I disabled collecting logs about my actions and sending this data to the same Google.  
For example, even in your tool, I have replaced all occurrences of the address `translate.google.com` on this one `translate.google.com.vn` ))  This means that using Google translation through your tool , no information is indexed to my Google profile! Because I'm not logged in to cookies for a Vietnamese domain.
Still, I think, if possible, make a custom setting to enable auto-reset of caches when closing the tab.. Or, for example, on a timer - after 10 minutes.
And I certainly have no complaints and do not demand anything, but just asked this question)
Thank you again for a great tool!!!

**Comment by FilipePS** [comment]
Persistent translation caching allows you to translate faster when visiting the same website (sometimes even when visiting different websites). It reduces the overhead of translation services and prevents them from temporarily blocking your IP address. In a way it also increases your privacy as fewer requests will be made to translation services.

**Comment by FilipePS** [comment]
One thing I forgot to mention. If you translate a site in an incognito window, it will not be cached on disk.

**Comment by FilipePS** [comment, marked as answer]
Currently disk caching is disabled by default.

**Reply by FiLinX** [reply]
I really, really appreciate it! 
Today it is difficult to meet such responsiveness of creators!
Thank you and Success in your Life!!)
